contrib/babelfishpg_common/src/spatialtypes.c

```diff
@@ -0,0 +1,134 @@
+/*-------------------------------------------------------------------------
+ *
+ * spatialtypes.c
+ *    Functions for the type "geometry" and "geography".
+ *
+ *-------------------------------------------------------------------------
+ */
+#ifdef ENABLE_SPATIAL_TYPES
+
+#include "postgres.h"
+#include "fmgr.h"
+#include "utils/geo_decls.h"
+#include "utils/builtins.h"
+
+/* This is copy of a struct from POSTGIS so that we could store and use the following values directly */
+typedef struct
+{
+    uint32_t size; /* For PgSQL use only, use VAR* macros to manipulate. */
+    uint8_t srid[3]; /* 24 bits of SRID */
+    uint8_t gflags; /* HasZ, HasM, HasBBox, IsGeodetic */
+    uint8_t data[1]; /* See gserialized.txt */
+} GSERIALIZED;
+
+typedef Datum (*LWGEOM_in_t)(PG_FUNCTION_ARGS);
+static LWGEOM_in_t LWGEOM_in_p;
+
+typedef Datum (*gserialized_set_srid_t)(PG_FUNCTION_ARGS);
+static gserialized_set_srid_t gserialized_set_srid_p;
+
+typedef Datum (*ST_FLIP_COORD_t)(PG_FUNCTION_ARGS);
+static ST_FLIP_COORD_t ST_FLIP_COORD_p;
+
+typedef Datum (*lwgeom_x_t)(PG_FUNCTION_ARGS);
+static lwgeom_x_t lwgeom_x_p;
+
+PG_FUNCTION_INFO_V1(LWGEOM_in);
+PG_FUNCTION_INFO_V1(geography_in);
+
+Datum
+LWGEOM_in(PG_FUNCTION_ARGS)
+{
+    Datum geom_datum;
+    GSERIALIZED *geom; /* Used to Store the bytes in the Format which is stored in PostGIS */
+
+    /* Ensure LWGEOM_in_p is properly initialized before using it */
+    if (LWGEOM_in_p == NULL) LWGEOM_in_p = (LWGEOM_in_t) load_external_function("$libdir/postgis-3", "LWGEOM_in", true, NULL);
+
+    /* Call the LWGEOM_in function via the function pointer */
+    geom_datum = LWGEOM_in_p(fcinfo);
+
+    geom = (GSERIALIZED*)PG_DETOAST_DATUM(geom_datum);
+
+    /* check if it is a 2-D point type */
+    if(*((uint32_t*)geom->data) != 1) ereport(ERROR,
+                                        (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                                        errmsg("Currently only Point type is supported")));
```

Comment:
Let's be consistent with the error message. Make sure that we are throwing the same error from both SQL functions and C functions for same scenarios.

Reply:
Done